apis/python/src/tiledbsoma/_util.py

```diff
@@ -546,24 +553,18 @@ def _set_coord(
 
     # Note: slice(None, None) matches the is_slice_of part, unless we also check
     # the dim-type part.
-    if is_slice_of(coord, np.datetime64) and pa.types.is_timestamp(dim.type):
+    if (
+        is_slice_of(coord, np.datetime64)
```

Comment:
Might be simpler if `is_slice_of` could be modified to take a tuple? So here you could write

```
  if (is_slice_of(coord, (np.datetime64, pa.TimestampScalar, int)):
```

Reply:
`is_slice_of` is implemented in somacore. I can update the code in somacore to also support tuples and then make a follow-up PR for tiledbsoma-py.